proxy/src/main.rs

```diff
@@ -108,6 +123,37 @@ async fn main() -> anyhow::Result<()> {
         )
         .get_matches();
 
+    if let Some(dir) = arg_matches.value_of("datadir") {
+        // change into the data directory.
+        println!("Working in {}", dir);
+        std::env::set_current_dir(PathBuf::from(dir))?;
+    }
+
+    if arg_matches.is_present("daemonize") {
+        println!("daemonizing...");
+
+        // There should'n be any logging to stdin/stdout. Redirect it to the main log so
+        // that we will see any accidental manual fprintf's or backtraces.
+        let log_file = zenith_utils::logging::init("proxy.log", true)?;
+        let stdout = log_file.try_clone().unwrap();
+        let stderr = log_file;
+
+        let daemonize = daemonize::Daemonize::new()
+            .pid_file("proxy.pid")
+            .working_directory(Path::new("."))
+            .stdout(stdout)
+            .stderr(stderr);
+
+        // XXX: The parent process should exit abruptly right after
+        // it has spawned a child to prevent coverage machinery from
+        // dumping stats into a `profraw` file now owned by the child.
+        // Otherwise, the coverage data will be damaged.
+        match daemonize.exit_action(|| exit_now(0)).start() {
+            Ok(_) => println!("Success, daemonized"),
+            Err(e) => anyhow::bail!("Error, {}", e),
+        }
```

Comment:
copied from safekeeper. If this PR merges I'll consider making this a zenith util.